### ISSUE TITLE
feat: combinations/permutations like in itertools for homogeneous tuples

### ISSUE THREAD

**Issue by dzmitry-lahoda** [libsugar/tuplers#10]
itertools has combinations and permutations, please make them for tuples.

why cannot just use itertools? 
because itertools work only on iterators, not fixed size arrays/slices.

i need to keep size know after combinations.

**Comment by 2A5F** [comment]
Can you give an example?

**Comment by dzmitry-lahoda** [comment]
using itertools:

```rust
    let sessions = [alice_session, bob_session]
        .iter()
        .permutations(2)
        .map(|x| (*x[0], *x[1]))
        .collect::<Vec<_>>();
    let sides = [Side::Ask, Side::Bid]
        .iter()
        .permutations(2)
        .map(|x| (*x[0], *x[1]))
        .collect::<Vec<_>>();
```

with tuples I can drop some lines

**Comment by 2A5F** [comment]
Included in v1.16.0
